app/image_recognizer.py

```diff
@@ -0,0 +1,154 @@
+
+    class MissingModelsLinksFile(Exception):
+        pass
+
+    class InvalidModelsLinksFile(Exception):
+        pass
+
+    class InvalidModelLink(Exception):
+        pass
+
+    class CustomModelNotFound(Exception):
+        pass
+
+    @dataclass
+    class Result:
+        color_format: str = ""
+        img_width: int = 0
+        img_height: int = 0
+        color_format_confidence: float = 0.0
+        resolution_confidence: float = 0.0
+
+    def __init__(
+        self,
+        resolution_model_path: str = DEFAULT_RESOLUTION_MODEL_PATH,
+        color_format_model_path: str = DEFAULT_COLOR_FORMAT_MODEL_PATH,
+        resolution_model_img_width: int = DEFAULT_RESOLUTION_MODEL_IMG_WIDTH,
+        resolution_model_img_height: int = DEFAULT_RESOLUTION_MODEL_IMG_HEIGHT,
+        color_format_model_img_width: int = DEFAULT_COLOR_FORMAT_MODEL_IMG_WIDTH,
+        color_format_model_img_height: int = DEFAULT_COLOR_FORMAT_MODEL_IMG_HEIGHT,
+    ):
+        """Create class object, set all instance variables, download defaults keras models if needed,
+         create ResolutionFinder and ColorFormatFinder objects
+
+        Args:
+            resolution_model_path (str, optional): Path to resolution keras model file.
+             Defaults to DEFAULT_RESOLUTION_MODEL_PATH.
+            color_format_model_path (str, optional): Path to color format keras model file.
+             Defaults to DEFAULT_COLOR_FORMAT_MODEL_PATH.
+            resolution_model_img_width (int, optional): Resolution keras model image width.
+             Defaults to DEFAULT_RESOLUTION_MODEL_IMG_WIDTH.
+            resolution_model_img_height (int, optional): Resolution keras model image height.
+             Defaults to DEFAULT_RESOLUTION_MODEL_IMG_HEIGHT.
+            color_format_model_img_width (int, optional): Color format keras model image width.
+             Defaults to DEFAULT_COLOR_FORMAT_MODEL_IMG_WIDTH.
+            color_format_model_img_height (int, optional): Color format keras model image height.
+             Defaults to DEFAULT_COLOR_FORMAT_MODEL_IMG_HEIGHT.
+
+        Raises:
+            self.CustomModelNotFound: Custom model not found
+        """
+        self.resolution_model_path = resolution_model_path
+        self.color_format_model_path = color_format_model_path
+        self.resolution_model_img_width = resolution_model_img_width
+        self.resolution_model_img_height = resolution_model_img_height
+        self.color_format_model_img_width = color_format_model_img_width
+        self.color_format_model_img_height = color_format_model_img_height
+
+        if (not exists(self.resolution_model_path)) or (not exists(self.color_format_model_path)):
+            if (self.resolution_model_path == self.DEFAULT_RESOLUTION_MODEL_PATH) and (
+                self.color_format_model_path == self.DEFAULT_COLOR_FORMAT_MODEL_PATH
+            ):
+                self.download_default_models()
```

Comment:
What if one path is not default? Then second default model wouldnt be downloaded and exception would be raised

Reply:
Yes, that's true, I will change it tomorrow.

Reply:
Done.